// ace-editor-builds/src/theme-snappy-light.js
define("ace/theme/snappy-light",["require","exports","module","ace/lib/dom"], function(require, exports, module) {

exports.isDark = true;
exports.cssClass = "ace-snappy-light";
exports.cssText = ".ace-snappy-light .ace_gutter {\
  background: #ffffff;\
  color: rgb(170,170,170)\
}\
.ace_gutter-cell {\
	padding-left:14px !important;\
	border-left: 6px solid #e67e22 !important\
}\
\
.ace-snappy-light .ace_print-margin {\
  width: 1px;\
  background: #e8e8e8\
}\
\
.ace_dark .ace_fold-widget {\
	background-image:url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAANElEQVR42mWKsQ0AMAzC8ixLlrzQjzmBiEjp0A6WwBCSPgKAXoLkqSot7nN3yMwR7pZ32NzpKkVoDBUxKAAAAABJRU5ErkJggg==')\
}\
.ace-snappy-light {\
  background-color: #ffffff;\
  color: #555555\
}\
\
.ace-snappy-light .ace_cursor {\
  color: #444444\
}\
\
.ace-snappy-light .ace_marker-layer .ace_selection {\
  background: #808dd3\
}\
\
.ace-snappy-light.ace_multiselect .ace_selection.ace_start {\
  box-shadow: 0 0 3px 0px #ffffff;\
  border-radius: 2px\
}\
\
.ace-snappy-light .ace_marker-layer .ace_step {\
  background: rgb(198, 219, 174)\
}\
\
.ace-snappy-light .ace_marker-layer .ace_bracket {\
  margin: -1px 0 0 -1px;\
  border: 1px solid #3b3a32\
}\
\
.ace-snappy-light .ace_marker-layer .ace_active-line {\
  background: #eeeeee\
}\
\
.ace-snappy-light .ace_gutter-active-line {\
  background-color: #eeeeee\
}\
\
.ace-snappy-light .ace_marker-layer .ace_selected-word {\
  border: 1px solid #808dd3\
}\
\
.ace-snappy-light .ace_fold {\
  background-color: #808dd3;\
  border-color: #555555\
}\
\
.ace-snappy-light .ace_keyword {\
  color: #da564a\
}\
\
.ace-snappy-light .ace_keyword.ace_other.ace_unit {\
  color: #4ea1df\
}\
\
.ace-snappy-light .ace_constant.ace_language {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_constant.ace_numeric {\
  color: #4ea1df\
}\
\
.ace-snappy-light .ace_constant.ace_character {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_constant.ace_other {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_support.ace_function {\
  color: #606aa1\
}\
\
.ace-snappy-light .ace_support.ace_constant {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_support.ace_constant.ace_property-value {\
  color: #4ea1df\
}\
\
.ace-snappy-light .ace_support.ace_class {\
  font-style: italic;\
  color: #f66153\
}\
\
.ace-snappy-light .ace_support.ace_type {\
  font-style: italic;\
  color: #f66153\
}\
\
.ace-snappy-light .ace_storage {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_storage.ace_type {\
  color: #4ea1df\
}\
\
.ace-snappy-light .ace_invalid {\
  color: #f8f8f0;\
  background-color: #00a8c6\
}\
\
.ace-snappy-light .ace_invalid.ace_deprecated {\
  color: #f8f8f0;\
  background-color: #00a8c6\
}\
\
.ace-snappy-light .ace_string {\
  color: #4ea1df\
}\
\
.ace-snappy-light .ace_comment {\
  color: #bbbbbb\
}\
\
.ace-snappy-light .ace_variable {\
  color: #808dd3\
}\
\
.ace-snappy-light .ace_variable.ace_parameter {\
  font-style: italic\
}\
\
.ace-snappy-light .ace_entity.ace_other.ace_attribute-name {\
  color: #f66153\
}\
\
.ace-snappy-light .ace_entity.ace_name.ace_function {\
  color: #808dd3\
}\
\
.ace-snappy-light .ace_entity.ace_name.ace_tag {\
  color: #4ea1df\
}";

var dom = require("../lib/dom");
dom.importCssString(exports.cssText, exports.cssClass);
});
