// ace-editor-builds/src/theme-rubyblue.js
define("ace/theme/rubyblue",["require","exports","module","ace/lib/dom"], function(require, exports, module) {

exports.isDark = true;
exports.cssClass = "ace-rubyblue";
exports.cssText = ".ace-rubyblue .ace_gutter {\
  background: #1F4661;\
  color: #fff\
}\
\
.ace_gutter-cell {\
	padding-left:14px;\
	border-right: 6px solid #3E7087\
}\
.ace-rubyblue .ace_print-margin {\
  width: 1px;\
  background: #e8e8e8\
}\
\
.ace-rubyblue {\
  background-color: #121E31;\
  color: #FFFFFF\
}\
\
.ace-rubyblue .ace_cursor {\
  color: #FFFFFF\
}\
\
.ace-rubyblue .ace_marker-layer .ace_selection {\
  background: #38566F\
}\
\
.ace-rubyblue.ace_multiselect .ace_selection.ace_start {\
  box-shadow: 0 0 3px 0px #121E31;\
  border-radius: 2px\
}\
\
.ace-rubyblue .ace_marker-layer .ace_step {\
  background: rgb(198, 219, 174)\
}\
\
.ace-rubyblue .ace_marker-layer .ace_bracket {\
  margin: -1px 0 0 -1px;\
  border: 1px solid #404040\
}\
\
.ace-rubyblue .ace_marker-layer .ace_active-line {\
  background: #253E5A\
}\
\
.ace-rubyblue .ace_gutter-active-line {\
  background-color: #253E5A\
}\
\
.ace-rubyblue .ace_marker-layer .ace_selected-word {\
  border: 1px solid #38566F\
}\
\
.ace-rubyblue .ace_fold {\
  background-color: #F8BB00;\
  border-color: #FFFFFF\
}\
\
.ace-rubyblue .ace_keyword {\
  color: #F8BB00\
}\
\
.ace-rubyblue .ace_keyword.ace_operator {\
  color: #8AA6C1\
}\
\
.ace-rubyblue .ace_constant {\
  color: #8AA6C1\
}\
\
.ace-rubyblue .ace_constant.ace_numeric {\
  color: #EDDD3D\
}\
\
.ace-rubyblue .ace_constant.ace_other {\
  color: #B53B3C\
}\
\
.ace-rubyblue .ace_support.ace_function {\
  color: #B43D3D\
}\
\
.ace-rubyblue .ace_string.ace_regexp {\
  color: #CA4344\
}\
\
.ace-rubyblue .ace_comment {\
  font-style: italic;\
  color: #428BDD\
}\
\
.ace-rubyblue .ace_variable {\
  color: #8AA6C1\
}\
\
.ace-rubyblue .ace_variable.ace_parameter {\
  font-style: italic;\
  color: #FFFFFF\
}";

var dom = require("../lib/dom");
dom.importCssString(exports.cssText, exports.cssClass);
});
