// ace-editor-builds/src/theme-wwdc.js
define("ace/theme/wwdc",["require","exports","module","ace/lib/dom"], function(require, exports, module) {

exports.isDark = true;
exports.cssClass = "ace-wwdc";
exports.cssText = ".ace-wwdc .ace_gutter {\
  background: #292B37;\
  color: rgb(148,149,155)\
}\
\
.ace-wwdc .ace_print-margin {\
  width: 1px;\
  background: #e8e8e8\
}\
\
.ace-wwdc {\
  background-color: #292B37;\
  color: #FFFFFF\
}\
\
.ace-wwdc .ace_cursor {\
  color: #F8F8F0\
}\
\
.ace-wwdc .ace_marker-layer .ace_selection {\
  background: #49483E\
}\
\
.ace-wwdc.ace_multiselect .ace_selection.ace_start {\
  box-shadow: 0 0 3px 0px #292B37;\
  border-radius: 2px\
}\
\
.ace-wwdc .ace_marker-layer .ace_step {\
  background: rgb(198, 219, 174)\
}\
\
.ace-wwdc .ace_marker-layer .ace_bracket {\
  margin: -1px 0 0 -1px;\
  border: 1px solid #3B3A32\
}\
\
.ace-wwdc .ace_marker-layer .ace_active-line {\
  background: #3E3D32\
}\
\
.ace-wwdc .ace_gutter-active-line {\
  background-color: #3E3D32\
}\
\
.ace-wwdc .ace_marker-layer .ace_selected-word {\
  border: 1px solid #49483E\
}\
\
.ace-wwdc .ace_fold {\
  background-color: #8884C5;\
  border-color: #FFFFFF\
}\
\
.ace-wwdc .ace_keyword {\
  color: #D28E5D\
}\
\
.ace-wwdc .ace_constant.ace_language {\
  color: #AE81FF\
}\
\
.ace-wwdc .ace_constant.ace_numeric {\
  color: #AE81FF\
}\
\
.ace-wwdc .ace_constant.ace_character {\
  color: #AE81FF\
}\
\
.ace-wwdc .ace_constant.ace_other {\
  color: #AE81FF\
}\
\
.ace-wwdc .ace_support.ace_function {\
  color: #8884C5\
}\
\
.ace-wwdc .ace_support.ace_constant {\
  color: #8884C5\
}\
\
.ace-wwdc .ace_support.ace_class {\
  font-style: italic;\
  color: #8884C5\
}\
\
.ace-wwdc .ace_support.ace_type {\
  font-style: italic;\
  color: #8884C5\
}\
\
.ace-wwdc .ace_storage {\
  color: #00ABA4\
}\
\
.ace-wwdc .ace_storage.ace_type {\
  font-style: italic;\
  color: #00ABA4\
}\
\
.ace-wwdc .ace_invalid {\
  color: #F8F8F0;\
  background-color: #F92672\
}\
\
.ace-wwdc .ace_invalid.ace_deprecated {\
  color: #F8F8F0;\
  background-color: #AE81FF\
}\
\
.ace-wwdc .ace_string {\
  color: #DC3B3B\
}\
\
.ace-wwdc .ace_comment {\
  color: #52BC57\
}\
\
.ace-wwdc .ace_variable {\
  color: #8884C5\
}\
\
.ace-wwdc .ace_variable.ace_parameter {\
  font-style: italic;\
  color: #D28E5D\
}\
\
.ace-wwdc .ace_entity.ace_other.ace_attribute-name {\
  color: #B73999\
}\
\
.ace-wwdc .ace_entity.ace_name.ace_function {\
  color: #8884C5\
}\
\
.ace-wwdc .ace_entity.ace_name.ace_tag {\
  color: #F92672\
}";
var dom = require("../lib/dom");
dom.importCssString(exports.cssText, exports.cssClass);
});
